pkg/propertiesformatter/yamlFormatter.go

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package propertiesformatter
+
+import (
+	"strings"
+
+	"github.com/galasa-dev/cli/pkg/galasaapi"
+	"gopkg.in/yaml.v3"
+)
+
+// -----------------------------------------------------
+// Yaml format.
+const (
+	YAML_FORMATTER_NAME = "yaml"
+)
+
+type PropertyYamlFormatter struct {
+}
+
+func NewPropertyYamlFormatter() PropertyFormatter {
+	return new(PropertyYamlFormatter)
+}
+
+func (*PropertyYamlFormatter) GetName() string {
+	return YAML_FORMATTER_NAME
+}
+
+func (*PropertyYamlFormatter) FormatProperties(cpsProperties []galasaapi.CpsProperty) (string, error) {
+	var result string = ""
+	var err error = nil
+	buff := strings.Builder{}
+	totalProperties := len(cpsProperties)
+
+	if totalProperties > 0 {
+		buff.WriteString("apiVersion: galasa-dev/v1alpha1\n")
+	}
+	for index, property := range cpsProperties {
+		propertyString := ""
+
+		if index > 0 {
+			propertyString += "---\n"
+		}
+		
+		var yamlRepresentationBytes []byte
+		yamlRepresentationBytes, err = yaml.Marshal(property)
+		if err == nil {
+			yamlStr := string(yamlRepresentationBytes)
+			propertyString += yamlStr
+		}
+
+		buff.WriteString(propertyString)
+	}
+
+	result = buff.String()
+	return result, err
+}
+
+func (*PropertyYamlFormatter) FormatNamespaces(namespaces []galasaapi.Namespace) (string, error) {
+	var result string = ""
+	var err error = nil
+	buff := strings.Builder{}
+	totalProperties := len(namespaces)
+
+	if totalProperties > 0 {
+		buff.WriteString("apiVersion: galasa-dev/v1alpha1\n")
+	}
+	for index, namespace := range namespaces {
+		propertyString := ""
+
+		if index > 0 {
+			propertyString += "---\n"
+		}
+		
+		var yamlRepresentationBytes []byte
+		yamlRepresentationBytes, err = yaml.Marshal(namespace)
+		if err == nil {
+			yamlStr := string(yamlRepresentationBytes)
+			propertyString += yamlStr
+		}
+
+		buff.WriteString(propertyString)
+	}
+
+	result = buff.String()
+	return result, err
+}
```

Comment:
We could refactor this to avoid the code duplication by converting the properties/namespaces into a `[]string` and then we can use a common helper function to build the formatted YAML - just a thought!

Reply:
Definitely a good idea! I will discuss with Mike about having `galasaapi.Namespace` and `galasaapi.CpsProperty `implement a common interface, like you mentioned!

Reply:
Sounds good, though it might not be possible since those structs are auto-generated.

Reply:
Yup, we played about with possible solutions but came to the conclusion that it would be best to leave them as they are, because things like typecasting and extracting the different information from both structs proved difficult. Also we are likely to change how we currently do the formatting in the future.